src/v3/src/types/widget.ts

```diff
@@ -67,6 +75,21 @@ export type WidgetResetArgs = {
   skipValidation?: boolean;
 };
 
+export type RegistrationSchemaCallbackV3 = (schema: RegistrationElementSchema[]) => void;
+export type RegistrationDataCallbackV3 = (postData: IdxActionParams) => void;
+export type RegistrationOptionsV3 = Modify<RegOptions, {
+  parseSchema?: (
+    schema: RegistrationElementSchema[],
+    onSuccess: RegistrationSchemaCallbackV3,
+    onFailure: RegistrationErrorCallback
+  ) => void;
+  preSubmit?: (
+    postData: IdxActionParams,
+    onSuccess: RegistrationDataCallbackV3,
+    onFailure: RegistrationErrorCallback
+  ) => void;
+}>;
```

Comment:
I had to modify this type because it looked like this was originally made for v1 (the current structure did not match v2 at all).